frontend/controller/actions/group-kv.js

```diff
@@ -30,12 +30,26 @@ export default (sbp('sbp/selectors/register', {
     console.info('group key-value store data loaded!')
   },
   'gi.actions/group/kv/fetchLastLoggedIn': async ({ contractID }: { contractID: string }) => {
-    return (await sbp('chelonia/kv/get', contractID, KV_KEYS.LAST_LOGGED_IN))?.data || Object.create(null)
+    const kvData = await sbp('chelonia/kv/get', contractID, KV_KEYS.LAST_LOGGED_IN)
+    // kvData could be falsy if the server returns 404
+    if (kvData) {
+      // Note: this could throw an exception if there's a signature or decryption
+      // issue
+      return kvData.data
+    }
+    return Object.create(null)
   },
   'gi.actions/group/kv/loadLastLoggedIn': ({ contractID }: { contractID: string }) => {
     return sbp('okTurtles.eventQueue/queueEvent', KV_QUEUE, async () => {
       const data = await sbp('gi.actions/group/kv/fetchLastLoggedIn', { contractID })
       sbp('okTurtles.events/emit', NEW_LAST_LOGGED_IN, [contractID, data])
+
+      // If running in a SW, update lastLoggedIn record. This is comparable to
+      // what the browser does when receiving the NEW_LAST_LOGGED_IN event
+      if (typeof WorkerGlobalScope === 'function') {
+        const rootState = sbp('state/vuex/state')
+        rootState.lastLoggedIn[contractID] = data
+      }
```

Comment:
- Can you add a comment explaining why this code is being done?
- If it needs to be done shouldn't it be done as part of all the KV code? (e.g. identity.js and any other keys?)